src/frontend.rs

```diff
@@ -100,17 +100,171 @@ impl NotificationExt for Notification {
+trait TrayIconTrait {
+    /// Initialize the tray icon
+    fn init() -> Result<GenericTrayIcon, ()>;
+    /// Set the sender to send inputs to the app
+    #[cfg(all(unix, not(target_os = "macos")))]
+    fn set_sender(&mut self, sender: AsyncComponentSender<App>);
+}
+
+/// Tray icon for Linux excluding macOS
+#[cfg(all(unix, not(target_os = "macos")))]
+struct LinuxTrayIcon {
+    icon: ksni::Icon,
+    sender: Option<AsyncComponentSender<App>>,
+}
+
+#[cfg(all(unix, not(target_os = "macos")))]
+impl ksni::Tray for LinuxTrayIcon {
+    fn id(&self) -> String {
+        env!("CARGO_PKG_NAME").to_string()
+    }
+
+    fn icon_name(&self) -> String {
+        "Space Acres".to_string()
+    }
+
+    fn title(&self) -> String {
+        "Space Acres".to_string()
+    }
+
+    fn icon_pixmap(&self) -> Vec<ksni::Icon> {
+        vec![self.icon.clone()]
+    }
+
+    fn tool_tip(&self) -> ksni::ToolTip {
+        ksni::ToolTip {
+            title: "Space Acres".to_string(),
+            ..Default::default()
+        }
+    }
+
+    fn menu(&self) -> Vec<ksni::MenuItem<Self>> {
+        use ksni::menu::*;
+
+        vec![
+            StandardItem {
+                label: T.tray_icon_open().to_string(),
+                icon_data: self.icon.data.clone(),
+                activate: Box::new(|this: &mut Self| {
+                    if let Some(sender) = this.sender.as_ref() {
+                        sender.input(AppInput::ShowWindow);
+                    }
+                }),
+                ..Default::default()
+            }
+            .into(),
+            StandardItem {
+                label: T.tray_icon_close().to_string(),
+                icon_data: self.icon.data.clone(),
+                activate: Box::new(|this: &mut Self| {
+                    if let Some(sender) = this.sender.as_ref() {
+                        sender.input(AppInput::HideWindow);
+                    }
+                }),
+                ..Default::default()
+            }
+            .into(),
+        ]
+    }
+}
+
+/// Tray icon for Windows and macOS
+#[cfg(any(target_os = "windows", target_os = "macos"))]
+struct OtherTrayIcon {
+    _inner: tray_icon::TrayIcon,
+}
+
+/// Variant for tray icons
+enum PlatformTrayIcon {
+    #[cfg(all(unix, not(target_os = "macos")))]
+    Linux(LinuxTrayIcon),
+    #[cfg(any(target_os = "windows", target_os = "macos"))]
+    Other { _inner: OtherTrayIcon },
+}
+
+/// Generic tray icon wrapper
+struct GenericTrayIcon {
+    _inner: PlatformTrayIcon,
+}
+
+impl TrayIconTrait for GenericTrayIcon {
+    fn init() -> Result<Self, ()> {
+        let icon_img = image::load_from_memory_with_format(ICON, image::ImageFormat::Png)
+            .expect("Statically correct image; qed")
+            .to_rgba8();
+
+        let (width, height) = icon_img.dimensions();
+
+        #[cfg(all(unix, not(target_os = "macos")))]
+        {
+            Ok(Self {
+                _inner: PlatformTrayIcon::Linux(LinuxTrayIcon {
```

Comment:
This is not sufficient to define a struct, it will not do anything. Just like for Windows and macOS icons are supposed to be driven to handle events, but you do not do that, so it will never appear in Linux. See `ksni::TrayService` in the official docs on how to actually make it do something.

This is why I wanted to abstract away implementation completely and expose the same API of driving with events in a command.